docs/src/developer/plan.rst

```diff
@@ -0,0 +1,97 @@
+***************
+Source code map
+***************
+
+This document will walk the path of an orion experiment through the
+code. Not every detail is explained, but there are ample links to the
+classes and methods involved if you want to dig further in a certain
+section.
+
+Departure
+---------
+
+You start an experience by running ``orion hunt <script> <params>``.
+
+The code in :py:func:`orion.core.cli.main` will parse the command line
+arguments and route to :py:func:`orion.core.cli.hunt.main`.
+
+The command line arguments are passed to
+:py:func:`orion.core.io.experiment_builder.build_from_args`. This will
+massage the parsed command line arguments and merge that configuration
+with the config file and the defaults with various helpers from
+:py:mod:`orion.core.io.resolve_config` to build the final
+configuration. The result is eventually handled off to
+:py:func:`orion.core.io.experiment_builder.create_experiment` to
```

Comment:
May be worth mentioning here, is the way the experiment configuration is fetched from the DB and compared with current one if already existing in DB. It also handles race conditions if any when multiple `hunt` calls try to create the same experiment at the same time.